Add development guide (#196)
pingcap/chaos-mesh: doc/DEVELOP.md

# Development documentation

> This tutorial will walk you through being a Chaos Mesh project developer from scratch, and cover something you're interested in.

Reviewer: ```suggestion
This guide prepares you for the development of Chaos Mesh from scratch. Before you get started, it is recommended you get familiar with the project through the following materials:

- [README](./README.md)
```

> The tutorial is listed here

Reviewer: ```suggestion
The development flow starts from [Set up your development environment]. After this, you can choose any of the following procedures to contribute:
```

> - [ ] Set up your development environment

Reviewer: ```suggestion
```

# Development documentation

> This tutorial will walk you through being a Chaos Mesh project developer from scratch, and cover something you're interested in.

Reviewer 2: Maybe `you may` is better than `you're` ?

> This section of the documentation is intended to get you to develop Chaos Mesh project. We'll cover things you need to know, from setup environment to add some kind of chaos.  

Reviewer 3: ```suggestion
This tutorial will walk you through being a Chaos Mesh project developer from scratch, and cover something you're interested in.
```
